### ISSUE TITLE
cpu load average is too high, but cpu usage is low

### ISSUE THREAD

**Issue by amkgi** [intel/kvm-sgx#18]
This looks like a bug. The load average increases, but it doesn't decrease when the workload is low. I have stopped all services and disable the network, but the la isn't decreasing. 
<img width="979" alt="Снимок экрана 2021-02-03 в 11 58 24" src="https://user-images.githubusercontent.com/26279705/106727490-97fd2800-661c-11eb-89dc-1f2d54335f51.png">

```
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           2  0.0  0.0      0     0 ?        S    Jan27   0:00 [kthreadd]
root           3  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [rcu_gp]
root           4  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [rcu_par_gp]
root           6  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/0:0H-kb]
root           8  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [mm_percpu_wq]
root           9  0.0  0.0      0     0 ?        S    Jan27   0:38  \_ [ksoftirqd/0]
root          10  0.0  0.0      0     0 ?        I    Jan27   8:00  \_ [rcu_sched]
root          11  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/0]
root          13  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/0]
root          14  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/1]
root          15  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/1]
root          16  0.0  0.0      0     0 ?        S    Jan27   0:28  \_ [ksoftirqd/1]
root          18  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/1:0H-kb]
root          19  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/2]
root          20  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/2]
root          21  0.0  0.0      0     0 ?        S    Jan27   0:21  \_ [ksoftirqd/2]
root          23  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/2:0H-kb]
root          24  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/3]
root          25  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/3]
root          26  0.0  0.0      0     0 ?        S    Jan27   0:17  \_ [ksoftirqd/3]
root          28  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/3:0H-kb]
root          29  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/4]
root          30  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/4]
root          31  0.0  0.0      0     0 ?        S    Jan27   0:29  \_ [ksoftirqd/4]
root          33  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/4:0H-kb]
root          34  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/5]
root          35  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/5]
root          36  0.0  0.0      0     0 ?        S    Jan27   0:15  \_ [ksoftirqd/5]
root          38  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/5:0H-kb]
root          39  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/6]
root          40  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/6]
root          41  0.0  0.0      0     0 ?        S    Jan27   0:21  \_ [ksoftirqd/6]
root          43  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/6:0H-kb]
root          44  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/7]
root          45  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/7]
root          46  0.0  0.0      0     0 ?        S    Jan27   0:19  \_ [ksoftirqd/7]
root          48  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/7:0H]
root          49  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/8]
root          50  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/8]
root          51  0.0  0.0      0     0 ?        S    Jan27   0:20  \_ [ksoftirqd/8]
root          53  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/8:0H-kb]
root          54  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/9]
root          55  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/9]
root          56  0.0  0.0      0     0 ?        S    Jan27   0:13  \_ [ksoftirqd/9]
root          58  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/9:0H-kb]
root          59  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/10]
root          60  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/10]
root          61  0.0  0.0      0     0 ?        S    Jan27   0:20  \_ [ksoftirqd/10]
root          63  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/10:0H-k]
root          64  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/11]
root          65  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/11]
root          66  0.0  0.0      0     0 ?        S    Jan27   0:16  \_ [ksoftirqd/11]
root          68  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/11:0H-k]
root          69  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/12]
root          70  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/12]
root          71  0.0  0.0      0     0 ?        S    Jan27   0:12  \_ [ksoftirqd/12]
root          73  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/12:0H-k]
root          74  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/13]
root          75  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/13]
root          76  0.0  0.0      0     0 ?        S    Jan27   0:19  \_ [ksoftirqd/13]
root          78  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/13:0H-k]
root          79  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/14]
root          80  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/14]
root          81  0.0  0.0      0     0 ?        S    Jan27   0:15  \_ [ksoftirqd/14]
root          83  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/14:0H-k]
root          84  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [cpuhp/15]
root          85  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [migration/15]
root          86  0.0  0.0      0     0 ?        S    Jan27   0:21  \_ [ksoftirqd/15]
root          88  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/15:0H-k]
root          89  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [kdevtmpfs]
root          90  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [netns]
root          91  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [rcu_tasks_rude_]
root          92  0.0  0.0      0     0 ?        S    Jan27   0:12  \_ [kauditd]
root          93  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [khungtaskd]
root          94  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [oom_reaper]
root          95  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [writeback]
root          96  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [kcompactd0]
root          97  0.0  0.0      0     0 ?        SN   Jan27   0:00  \_ [ksmd]
root          98  0.0  0.0      0     0 ?        SN   Jan27   0:21  \_ [khugepaged]
root         149  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kintegrityd]
root         150  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kblockd]
root         151  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [blkcg_punt_bio]
root         153  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [tpm_dev_wq]
root         154  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [md]
root         155  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [edac-poller]
root         156  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [devfreq_wq]
root         157  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [watchdogd]
root         160  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [ksgxswapd]
root         161  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [kswapd0]
root         164  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kthrotld]
root         174  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [acpi_thermal_pm]
root         175  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kmpath_rdacd]
root         176  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kaluad]
root         177  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [ipv6_addrconf]
root         178  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kstrp]
root         179  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [zswap-shrink]
root         190  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [charger_manager]
root         191  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/u33:0]
root         371  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [iscsi_eh]
root         372  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [iscsi_destroy]
root         385  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [rpciod]
root         386  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [xprtiod]
root         500  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_0]
root         501  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_0]
root         502  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [ixgbe]
root         503  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [ata_sff]
root         505  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [cryptd]
root         573  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_1]
root         574  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_1]
root         575  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_2]
root         576  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_2]
root         577  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_3]
root         578  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_3]
root         579  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_4]
root         580  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_4]
root         581  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_5]
root         582  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_5]
root         583  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_6]
root         584  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_6]
root         585  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_7]
root         586  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_7]
root         587  0.0  0.0      0     0 ?        S    Jan27   0:00  \_ [scsi_eh_8]
root         588  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [scsi_tmf_8]
root         601  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [poll_megasas0_s]
root         604  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/8:1H-kb]
root         630  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/5:1H-kb]
root         631  0.0  0.0      0     0 ?        S    Jan27   0:06  \_ [jbd2/sda3-8]
root         632  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [ext4-rsv-conver]
root         641  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/6:1H-ev]
root         700  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/10:1H-k]
root         715  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/9:1H-kb]
root         726  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/12:1H-e]
root         731  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/13:1H-k]
root         735  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/1:1H-kb]
root         736  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/11:1H-k]
root         737  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/0:1H-kb]
root         741  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/14:1H-k]
root         750  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/15:1H-k]
root         776  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/7:1H-kb]
root         778  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/4:1H-ev]
root         789  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/2:1H-kb]
root         831  0.0  0.0      0     0 ?        I<   Jan27   0:00  \_ [kworker/3:1H-kb]
root     2869392  0.0  0.0      0     0 ?        I    07:45   0:00  \_ [kworker/0:0-cgr]
root     2870902  0.0  0.0      0     0 ?        I    07:50   0:00  \_ [kworker/8:1-mm_]
root     2871432  0.0  0.0      0     0 ?        I    07:53   0:00  \_ [kworker/7:1-eve]
root     2871525  0.0  0.0      0     0 ?        I    07:53   0:00  \_ [kworker/10:1-ev]
root     2871641  0.0  0.0      0     0 ?        I    07:54   0:00  \_ [kworker/13:1-ev]
root     2871932  0.0  0.0      0     0 ?        I    07:56   0:00  \_ [kworker/14:1-mm]
root     2873348  0.0  0.0      0     0 ?        I    07:59   0:00  \_ [kworker/u32:3-e]
root     2873447  0.0  0.0      0     0 ?        I    07:59   0:00  \_ [kworker/15:11-c]
root     2873451  0.0  0.0      0     0 ?        I    07:59   0:00  \_ [kworker/9:11-ev]
root     2873461  0.0  0.0      0     0 ?        I    07:59   0:00  \_ [kworker/3:11-mm]
root     2873488  0.0  0.0      0     0 ?        I    07:59   0:00  \_ [kworker/4:0-mm_]
root     2873833  0.0  0.0      0     0 ?        I    07:59   0:00  \_ [kworker/1:4-mm_]
root     2874157  0.0  0.0      0     0 ?        I    08:00   0:00  \_ [kworker/10:0-ev]
root     2874449  0.0  0.0      0     0 ?        I    08:12   0:00  \_ [kworker/0:2-eve]
root     2876254  0.0  0.0      0     0 ?        I    08:20   0:00  \_ [kworker/2:1-mm_]
root     2876323  0.0  0.0      0     0 ?        I    08:20   0:00  \_ [kworker/13:2]
root     2876325  0.0  0.0      0     0 ?        I    08:20   0:00  \_ [kworker/4:2]
root     2876360  0.0  0.0      0     0 ?        I    08:21   0:00  \_ [kworker/8:0]
root     2876413  0.0  0.0      0     0 ?        I    08:27   0:00  \_ [kworker/5:0-mm_]
root     2876435  0.0  0.0      0     0 ?        I    08:28   0:00  \_ [kworker/1:1]
root     2876625  0.0  0.0      0     0 ?        I    08:30   0:00  \_ [kworker/6:2-cgr]
root     2876659  0.0  0.0      0     0 ?        I    08:31   0:00  \_ [kworker/9:1-eve]
root     2876843  0.0  0.0      0     0 ?        I    08:33   0:00  \_ [kworker/11:0-ev]
root     2876869  0.0  0.0      0     0 ?        I    08:35   0:00  \_ [kworker/u32:0-e]
root     2876896  0.0  0.0      0     0 ?        I    08:36   0:00  \_ [kworker/3:1]
root     2877058  0.0  0.0      0     0 ?        I    08:41   0:00  \_ [kworker/11:3-ev]
root     2877092  0.0  0.0      0     0 ?        I    08:43   0:00  \_ [kworker/12:1-ev]
root     2877177  0.0  0.0      0     0 ?        I    08:44   0:00  \_ [kworker/12:2-ev]
root     2877191  0.0  0.0      0     0 ?        I    08:45   0:00  \_ [kworker/2:0-eve]
root     2877223  0.0  0.0      0     0 ?        I    08:45   0:00  \_ [kworker/14:2-ev]
root     2877272  0.0  0.0      0     0 ?        I    08:45   0:00  \_ [kworker/u32:2]
root     2877273  0.0  0.0      0     0 ?        I    08:45   0:00  \_ [kworker/7:2]
root     2877275  0.0  0.0      0     0 ?        I    08:45   0:00  \_ [kworker/6:0-mm_]
root     2877277  0.0  0.0      0     0 ?        I    08:45   0:00  \_ [kworker/15:0-mm]
root     2877371  0.0  0.0      0     0 ?        I    08:46   0:00  \_ [kworker/5:1-eve]
root     2877465  0.0  0.0      0     0 ?        I    08:49   0:00  \_ [kworker/10:2-cg]
root     2877548  0.0  0.0      0     0 ?        I    08:50   0:00  \_ [kworker/12:0-ev]
root     2877555  0.0  0.0      0     0 ?        I    08:50   0:00  \_ [kworker/10:3-mm]
root     2877578  0.0  0.0      0     0 ?        I    08:54   0:00  \_ [kworker/u32:1]
root     2877581  0.0  0.0      0     0 ?        I    08:54   0:00  \_ [kworker/14:0]
root     2877586  0.0  0.0      0     0 ?        I    08:54   0:00  \_ [kworker/11:1-mm]
root     2877587  0.0  0.0      0     0 ?        I    08:54   0:00  \_ [kworker/9:0-eve]
root           1  0.0  0.0 193848  8616 ?        Ss   Jan27   6:27 /usr/lib/systemd/systemd --system --deserialize 18
root        4566  0.0  0.0  96584  4212 ?        Ss   Jan27   0:00 login -- alexey
alexey   2873054  0.0  0.0 116616  4400 tty1     Ss   07:58   0:00  \_ -bash
root     2873090  0.0  0.0 241344  7740 tty1     S    07:58   0:00      \_ sudo su
root     2873092  0.0  0.0 192044  4604 tty1     S    07:58   0:00          \_ su
root     2873093  0.0  0.0 117032  5200 tty1     S    07:58   0:02              \_ bash
root     2877595  0.0  0.0 155788  4220 tty1     R+   08:54   0:00                  \_ ps auxf
root     2877584  0.0  0.0 110220  1852 tty6     Ss+  08:54   0:00 /sbin/agetty --noclear tty6 linux
root     2877589  0.0  0.0  26396  2920 ?        Ss   08:54   0:00 /usr/lib/systemd/systemd-logind
dbus     2877590  0.0  0.0  66332  4196 ?        Ssl  08:54   0:00 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation
root     2877592  0.0  0.0 110220  2112 ttyS0    Ss+  08:54   0:00 /sbin/agetty --keep-baud 115200,38400,9600 ttyS0 vt220
```

```
[Wed Feb  3 08:59:30 2021] sysrq: Show backtrace of all active CPUs
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 4
[Wed Feb  3 08:59:30 2021] CPU: 4 PID: 2873093 Comm: bash Not tainted 5.8.0-rc4+ #1
[Wed Feb  3 08:59:30 2021] Hardware name:  /045M96, BIOS 2.2.3 09/27/2019
[Wed Feb  3 08:59:30 2021] Call Trace:
[Wed Feb  3 08:59:30 2021]  dump_stack+0x74/0x9a
[Wed Feb  3 08:59:30 2021]  nmi_cpu_backtrace.cold.8+0x14/0x52
[Wed Feb  3 08:59:30 2021]  ? lapic_can_unplug_cpu.cold.30+0x3c/0x3c
[Wed Feb  3 08:59:30 2021]  nmi_trigger_cpumask_backtrace+0xe6/0xeb
[Wed Feb  3 08:59:30 2021]  arch_trigger_cpumask_backtrace+0x19/0x20
[Wed Feb  3 08:59:30 2021]  sysrq_handle_showallcpus+0x17/0x20
[Wed Feb  3 08:59:30 2021]  __handle_sysrq.cold.16+0x49/0x113
[Wed Feb  3 08:59:30 2021]  write_sysrq_trigger+0x2f/0x35
[Wed Feb  3 08:59:30 2021]  proc_reg_write+0x40/0x90
[Wed Feb  3 08:59:30 2021]  __vfs_write+0x1b/0x40
[Wed Feb  3 08:59:30 2021]  vfs_write+0xab/0x1b0
[Wed Feb  3 08:59:30 2021]  ksys_write+0x61/0xe0
[Wed Feb  3 08:59:30 2021]  __x64_sys_write+0x1a/0x20
[Wed Feb  3 08:59:30 2021]  do_syscall_64+0x52/0x90
[Wed Feb  3 08:59:30 2021]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[Wed Feb  3 08:59:30 2021] RIP: 0033:0x7fd0ccfe5a00
[Wed Feb  3 08:59:30 2021] Code: Bad RIP value.
[Wed Feb  3 08:59:30 2021] RSP: 002b:00007ffe84a0ca58 EFLAGS: 00000246 ORIG_RAX: 0000000000000001
[Wed Feb  3 08:59:30 2021] RAX: ffffffffffffffda RBX: 0000000000000002 RCX: 00007fd0ccfe5a00
[Wed Feb  3 08:59:30 2021] RDX: 0000000000000002 RSI: 00007fd0cd907000 RDI: 0000000000000001
[Wed Feb  3 08:59:30 2021] RBP: 00007fd0cd907000 R08: 000000000000000a R09: 00007fd0cd908740
[Wed Feb  3 08:59:30 2021] R10: 00007fd0cd908740 R11: 0000000000000246 R12: 00007fd0cd2be400
[Wed Feb  3 08:59:30 2021] R13: 0000000000000002 R14: 0000000000000001 R15: 0000000000000000
[Wed Feb  3 08:59:30 2021] Sending NMI from CPU 4 to CPUs 0-3,5-15:
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 8 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 0 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 12 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 10 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 15 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 7 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 2 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 3 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 5 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 13 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 11 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 1 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 9 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 14 skipped: idling at intel_idle+0x8b/0x130
[Wed Feb  3 08:59:30 2021] NMI backtrace for cpu 6 skipped: idling at intel_idle+0x8b/0x130
```
This problem occurs on all servers with SGX. They are equipped with Intel Xeon E-2288G. I compiled and installed kernel 5.8.0-rc4-r3:
```
uname -a
Linux localhost 5.8.0-rc4+ #1 SMP Thu Oct 22 13:05:29 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
[config-5.8.0-rc4+.txt](https://github.com/intel/kvm-sgx/files/5917322/config-5.8.0-rc4%2B.txt)

From perf:
<img width="1440" alt="Снимок экрана 2021-02-03 в 12 17 30" src="https://user-images.githubusercontent.com/26279705/106726545-8e26f500-661b-11eb-84ce-58051713adef.png">
[Flame graph](https://drive.google.com/file/d/1-4CFmM4SNcB3zEeAdxs0a-ZCt5WM5c4L/view?usp=sharing)

After the servers are rebooted, load average will continue to grow for several days.
<img width="1322" alt="Снимок экрана 2021-02-03 в 13 12 10" src="https://user-images.githubusercontent.com/26279705/106732275-9124e400-6621-11eb-9fc8-0dc39eb3d403.png">

Servers: DELL  PowerEdge R340
Servers are used as compute nodes. But they aren't under load, there are no virtual machines.
Servers have kolla openstack and qemu-sgx 4.0.0-r3 installed.

**Comment by kaihuang** [comment]
Could you try latest master to see whether this issue still exists?

Also, does this issue apply to host SGX workloads, or apply to SGX workloads in KVM SGX guest?

**Comment by amkgi** [comment]
I was able to fix the problem of cpu la growth. On the sgx-v5.11-rc3 kernel, this problem is not observed, but when loading this kernel, the memory-backend-epc object drops out in qemu-sgx. This was also resolved by compiling only the qemu-sgx-v5.1.0-rc3 version. In other versions of qemu-sgx, this object also drops out.

```
[root@sgx-hypervisor /]# /usr/libexec/qemu-kvm -version
QEMU emulator version 5.1.50 (qemu-kvm-5.1.0-gc.sgx.el8.1)
Copyright (c) 2003-2020 Fabrice Bellard and the QEMU Project developers

[root@sgx-hypervisor /]# /usr/libexec/qemu-kvm -object help |grep memory
  memory-backend-epc
  memory-backend-file
  memory-backend-memfd
  memory-backend-ram

[root@sgx-hypervisor /]# uname -r
5.11.0-rc3+

[root@sgx-hypervisor2 /]# /usr/libexec/qemu-kvm -version
QEMU emulator version 4.0.0 (qemu-kvm-4.0.0-gc.sgx.el8.1)
Copyright (c) 2003-2019 Fabrice Bellard and the QEMU Project developers

[root@sgx-hypervisor2 /]# /usr/libexec/qemu-kvm -object help |grep memory
  memory-backend-file
  memory-backend-memfd
  memory-backend-ram

[root@sgx-hypervisor2 /]# uname -r
5.11.0-rc3+
```
